src/main/java/com/yahoo/bullet/storm/JoinBolt.java

```diff
@@ -138,19 +128,8 @@ protected AggregationQuery getQuery(String id, String queryString) {
         return null;
     }
 
-    private void initializeMetadata(Tuple tuple) {
-        String id = tuple.getString(TopologyConstants.ID_POSITION);
-        // Check if we have any buffered errors.
-        Clip error = bufferedErrors.get(id);
-        if (error != null) {
-            emit(error, tuple);
-            return;
-        }
-        // Otherwise buffer the metadata
-        bufferedMetadata.put(id, tuple);
-    }
-
     private void handleQuery(Tuple tuple) {
+        bufferedMetadata.put(tuple.getString(TopologyConstants.ID_POSITION), tuple);
```

Comment:
I think this approach is now confusing since we no longer do a join for the metadata. To keep it safe for future changes, how about we change QueryBolt#getQuery to take a Tuple (of TupleType#QUERY_TUPLE). That way, we don't have to put the metadata of a bad query into the map and remove it elsewhere.